lib/slime/parser/text_block.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Slime.Parser.TextBlock do
+  @moduledoc "
+  Utilities for parsing text blocks.
+  "
+
+  import Slime.Parser.Transform, only: [wrap_in_quotes: 1]
+
+  @doc """
+  Given a text block and its declaration indentation level (see below),
+  produces a string (or a dynamic EEx tuple) that can be inserted into the tree.
+
+  nested
+    | Text block
+       that spans over multiple lines
+  ---
+   ^
+  declaration indent
+  """
+  def render(lines, decl_indent, trailing_whitespace \\ "") do
+    [{first_line_indent, first_line, is_eex_line} | rest] = lines
+
+    text_indent = if first_line == "" do
+      [{indent, _, _} | _] = rest
```

Comment:
what about single `'` used for spacing, example similar to one in my work project:
```
p
  '
  span text
  ' .
```
I understand, `span<` could be used here, but it is a valid slim and should be supported (probably should add a test case for it)